### PR TITLE
refactor: account adding

### PR DESCRIPTION
**Pull request by JustSimplyKyle**
make the clone call more apparent(take the ownership instead of borrowing but instantly clones it in the function)

### CODE REVIEW

**Comment by JustSimplyKyle** [comment]
The part I'm not sure with is `retain`. Not sure if it's the desired effect.

File: app/native/src/api/storage/account_storage.rs [hunk 1]
```diff
@@ -28,17 +28,17 @@ impl StorageInstance<Self> for AccountStorage {
 }
 
 impl AccountStorage {
-    pub fn add_account(&mut self, account: &MinecraftAccount, is_main_account: Option<bool>) {
-        // Keep only the last account with the same UUID.
-        self.accounts.retain(|a| a.uuid != account.uuid);
-        self.accounts.push(account.clone());
+    pub fn add_account(&mut self, account: MinecraftAccount, is_main_account: bool) {
+        if self.accounts.iter().any(|x| x.uuid != account.uuid) {
```
Comment (SiongSng) [inline app/native/src/api/storage/account_storage.rs]: 這裡應該要可以覆蓋帳號資訊，假設使用者重新登入相同帳號的話，或者為了更新帳號中的資訊。

**Review by SiongSng — COMMENTED** [review]
這樣的做法的確更明瞭，只有一個小問題。

File: app/native/src/api/storage/account_storage.rs (hunk 1 above)
Comment (JustSimplyKyle) [reply app/native/src/api/storage/account_storage.rs]: 那這樣使用 retain 的確更好

**Review by SiongSng — APPROVED** [review]
LGTM

**Pull request merged**